test/test_SimulatedLinkamT95.py

```diff
@@ -0,0 +1,236 @@
+        self.assertNotEqual(status_bytes[6:10], '01b8')     # Temp != 44.0 C
+
+        # Finish off heating (and overshoot a bit)
+        linkam.process(5)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[6:10], '01b8')    # Temp == 44.0 C
+
+        # Should hold now, so temperature should not change
+        linkam.process(10)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x30')       # Auto-holding at limit
+        self.assertEqual(status_bytes[6:10], '01b8')    # Temp == 44.0 C
+
+    def test_simple_cool(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Set up to cool from 24.0 C to 4.0 C at 20.00 C/min
+        linkam.setRate('2000')
+        linkam.setLimit('40')
+        linkam.start()
+        linkam.process(0)
+
+        # Cool for almost a minute but not quite
+        linkam.process(59.5)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x20')           # Cooling status set
+        self.assertNotEqual(status_bytes[6:10], '0028')     # Temp != 4.0 C
+
+        # Finish off cooling (and overshoot a bit)
+        linkam.process(5)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[6:10], '0028')    # Temp == 4.0 C
+
+        # Should hold now, so temperature should not change
+        linkam.process(10)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x30')       # Auto-holding at limit
+        self.assertEqual(status_bytes[6:10], '0028')    # Temp == 4.0 C
+
+    def test_error_flag_overcool(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Ensure flag is not set
+        status_bytes = linkam.getStatus()
+        self.assertFalse(ord(status_bytes[1]) & 0x01)
+
+        # Set up to cool from 24.0 C to 4.0 C at 51.00 C/min
+        linkam.setRate('5100')
+        linkam.setLimit('40')
+        linkam.start()
+        linkam.process(0)
+
+        # Ensure flag is set after running a bit
+        linkam.process(0.1)
+        status_bytes = linkam.getStatus()
+        self.assertTrue(ord(status_bytes[1]) & 0x01)
+
+    def test_stop_command(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Set up to heat from 24.0 C to 44.0 C at 20.00 C/min
+        linkam.setRate('2000')
+        linkam.setLimit('440')
+        linkam.start()
+        linkam.process(0)
+
+        # Process for some time and then stop
+        linkam.process(10)
+        linkam.stop()
+        linkam.process(0)
```

Comment:
> After all `process(0)` could be implemented such that it is a no-op that doesn't result in going to the next cycle.

Hm... I would find this very surprising (to the point where I would raise an issue against it).

If I tell a system to process a cycle, I expect it to process a cycle (or throw an exception / return an error code). I suppose this comes back to something you said earlier:

> If time doesn't matter in a time based simulation then [...]

It's not really a time based simulation. It's a cycle based simulation. Time is just a parameter that gets passed along "to whomever it may concern". When I call `process(0)`, I'm saying: "Process a cycle. Oh, and by the way, if anyone asks, tell them dt is 0."

If that's the part that is confusing, maybe `process` itself isn't a good name?

But, as you say, it's questionable whether this is worth much effort. You would only call `process` manually in tests, anyway. It's called automatically during normal simulation. And while, normally, `dt` wouldn't be 0 since it loops and calculates a delta, there is nothing special about a delta of 0 either. The `dt` argument is -- and should only ever be -- used as a factor. Never as a direct condition (though things increased as a result of factoring in `dt`s may of course be conditions).